go/libraries/doltcore/sqle/diff_table.go

```diff
@@ -0,0 +1,324 @@
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/diff"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/doltdb"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/env"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/rowconv"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/schema"
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+const (
+	DoltDiffTablePrefix = "dolt_diff_"
+	toCommit            = "to_commit"
+	fromCommit          = "from_commit"
+)
+
+var _ sql.FilteredTable = (*DiffTable)(nil)
+
+type DiffTable struct {
+	name          string
+	dEnv          *env.DoltEnv
+	superSch      schema.Schema
+	joiner        *rowconv.Joiner
+	fromRoot      *doltdb.RootValue
+	toRoot        *doltdb.RootValue
+	fromCommitVal string
+	toCommitVal   string
+}
+
+func NewDiffTable(name string, dEnv *env.DoltEnv) *DiffTable {
+	ctx := context.TODO()
+	ssg := rowconv.NewSuperSchemaGen()
+	err := ssg.AddHistoryOfTable(ctx, name, dEnv.DoltDB)
+
+	// TODO: fix panics
+	if err != nil {
+		panic(err)
+	}
+
+	sch, err := ssg.GenerateSuperSchema(rowconv.NameKindPair{Name: "commit", Kind: types.StringKind})
+
+	// TODO: fix panics
+	if err != nil {
+		panic(err)
+	}
+
+	j, err := rowconv.NewJoiner(
+		[]rowconv.NamedSchema{{Name: diff.To, Sch: sch}, {Name: diff.From, Sch: sch}},
+		map[string]rowconv.ColNamingFunc{
+			diff.To:   toNamer,
+			diff.From: fromNamer,
+		})
+
+	// TODO: fix panics
+	if err != nil {
+		panic(err)
+	}
+
+	root1, err := dEnv.WorkingRoot(ctx)
+
+	if err != nil {
+		panic(err)
+	}
+
+	root2, err := dEnv.StagedRoot(ctx)
+
+	if err != nil {
+		panic(err)
+	}
+
+	return &DiffTable{name, dEnv, sch, j, root2, root1, "current", "HEAD"}
+}
+
+func (dt *DiffTable) Name() string {
+	return DoltDiffTablePrefix + dt.name
+}
+
+func (dt *DiffTable) String() string {
+	return DoltDiffTablePrefix + dt.name
+}
+
+func (dt *DiffTable) Schema() sql.Schema {
+	sqlSch, err := doltSchemaToSqlSchema(dt.Name(), dt.joiner.GetSchema())
+
+	// TODO: fix panics
+	if err != nil {
+		panic(err)
+	}
+
+	return sqlSch
+}
+
+func toNamer(name string) string {
+	return diff.To + "_" + name
+}
+
+func fromNamer(name string) string {
+	return diff.From + "_" + name
+}
+
+func (dt *DiffTable) Partitions(*sql.Context) (sql.PartitionIter, error) {
+	return &doltTablePartitionIter{}, nil
+}
+
+func tableData(ctx *sql.Context, root *doltdb.RootValue, tblName string, ddb *doltdb.DoltDB) (types.Map, schema.Schema, error) {
+	tbl, ok, err := root.GetTable(ctx, tblName)
+
+	if err != nil {
+		return types.EmptyMap, nil, err
+	}
+
+	var data types.Map
+	if !ok {
+		data, err = types.NewMap(ctx, ddb.ValueReadWriter())
+	} else {
+		data, err = tbl.GetRowData(ctx)
+	}
+
+	if err != nil {
+		return types.EmptyMap, nil, err
+	}
+
+	sch, err := tbl.GetSchema(ctx)
+
+	if err != nil {
+		return types.EmptyMap, nil, err
+	}
+
+	return data, sch, nil
+}
+
+func (dt *DiffTable) PartitionRows(ctx *sql.Context, part sql.Partition) (sql.RowIter, error) {
```

Comment:
You're not gracefully handling the case where there isn't a WHERE clause that specifies from_commit and to_commit

Reply:
default values for dt.fromRoot/dt.toRoot/dt.fromCommitVal/dt.toCommitVal are set in NewDiffTable

Reply:
verified with a test